sw/host/opentitantool/src/command/spi.rs

```diff
@@ -256,10 +282,30 @@ impl CommandDispatch for SpiProgram {
 
 /// Commands for interacting with a SPI EEPROM.
 #[derive(Debug, StructOpt, CommandDispatch)]
-pub enum SpiCommand {
+pub enum InternalSpiCommand {
     Sfdp(SpiSfdp),
     ReadId(SpiReadId),
     Read(SpiRead),
     Erase(SpiErase),
     Program(SpiProgram),
 }
+
+#[derive(Debug, StructOpt)]
+pub struct SpiCommand {
+    #[structopt(short, long, default_value = "0", help = "SPI bus")]
+    pub bus: u32,
```

Comment:
Specifying an integer is fine now.
I foresee that in in order to do SPI programming operations, the debugger will need to also know which pins are to be used for reset and bootstrap mode, as well as SPI protocol variant, all of which goes together with the spi bus number.  I imagine that this parameter will be changed into a string, which would be a key into a configuration section, declaring the bus number along with all the other details of the programming.  (We could still allow specifying a string "1" for instance, which would translate to bus 1 (with some default for what protocol, speed, and reset and bootloader pins to use.)
Once this PR is merged, I will rebase my configuration proposal to show what I mean.

Reply:
We can (and I would prefer) to add additional parameters at this layer, which would yield a command line like:

```
# Read 1K from an eeprom on spi bus 1 at 12MHz and 1.8 volts.
opentitantool --interface=whatever spi --bus=1 --speed=12000000 --voltage=1.8 read --length=0x400 foo.bin
```

Bootstrap, as it's own command can have parameters relavant to it.  To avoid duplicating the SPI parameter names, we can declare a "spi configuration" struct and use one of structopt's features to add that config struct to commands which require to know details of the SPI bus.